osquery/config/config.cpp

```diff
@@ -531,6 +554,58 @@ void stripConfigComments(std::string& json) {
   json = sink;
 }
 
+Expected<ConfigMap, Config::RestoreConfigError> Config::restoreConfigBackup() {
+  if (!FLAGS_config_enable_backup) {
+    return createError(Config::RestoreConfigError::BackupDisabled,
+                       "Config persistense is disabled.");
+  }
+
+  VLOG(1) << "Restoring backed up config from the database";
+  std::vector<std::string> keys;
+  ConfigMap config;
+
+  WriteLock lock(config_backup_mutex_);
```

Comment:
This is a public function and this mutex only prevents backups and restores from executing concurrently. Shouldn't we also prevent updates to the config from happening concurrently to this?

Reply:
That's what I was discussing with Alex. Right now, config.cpp contains 6 mutexes, I'm afraid we are not benefiting from that and it just complicates things. Let me fix it in the next PR if you don't mind.